### PR TITLE
Add REST implementation for Validator's `ValidatorIndex`

### PR DESCRIPTION
**Pull request by nalepae**
### What type of PR is this?
Feature

### Related gitHub issue:
https://github.com/prysmaticlabs/prysm/issues/11580

### What does this PR do? Why is it needed?
This adds an implementation for the `ValidatorIndex` validator client method by calling the Beacon Node REST API instead of the gRPC one.

### Comments for reviewers:
Some code (like `getStateValidator` method) has a lot of common with the not yet merged https://github.com/prysmaticlabs/prysm/pull/11671 pull request.

Once this or https://github.com/prysmaticlabs/prysm/pull/11671 is merged, I will modify the not merged one to create generic methods where it is needed.

### CODE REVIEW

File: validator/client/beacon-api/index.go
Comment (PatriceVignola) [inline validator/client/beacon-api/index.go:43]: I think this method should have its own method as well since it could be reused by other endpoints.

Comment (nalepae) [reply validator/client/beacon-api/index.go:43]: Yes, it's the point of my "Comment for reviewers" section in PR description.

> Comments for reviewers:
Some code (like getStateValidator method) has a lot if common with the not yet merged https://github.com/prysmaticlabs/prysm/pull/11671 pull request.
Once this or https://github.com/prysmaticlabs/prysm/pull/11671 is merged, I will modify the not merged one to create generic methods where it is needed.

File: validator/client/beacon-api/index.go
Comment (PatriceVignola) [inline validator/client/beacon-api/index.go:33]: This will panic if stateValidator.Data[0] is nil

File: validator/client/beacon-api/handlers_test.go
Comment (rkapka) [inline validator/client/beacon-api/handlers_test.go:96]: `pubKeys` is not needed - you don't use it in any meaningful way.

File: validator/client/beacon-api/index_test.go
Comment (rkapka) [inline validator/client/beacon-api/index_test.go:22]: please extract to package-level variable

File: validator/client/beacon-api/index.go
Comment (rkapka) [inline validator/client/beacon-api/index.go:68]: This is a duplicate of code in `getGenesis()`. Given that we will probably use this for every endpoint, please extract a private function. We can parameterize the endpoint name in the first error string.

Comment (PatriceVignola) [reply validator/client/beacon-api/index.go:68]: @nalepae I added a helper `getRestJsonResponse` in https://github.com/prysmaticlabs/prysm/pull/1171, you can either copy the implementation or wait until the PR is merged.

File: validator/client/beacon-api/handlers_test.go
Comment (nalepae) [reply validator/client/beacon-api/handlers_test.go:96]: `pubkeys` is used for example here:

https://github.com/nalepae/prysm/blob/0f929241f7c2c555ed9a07a9999ce96e121466aa/validator/client/beacon-api/index_test.go#L39

If, at the same time, I comment this line `[]string{stringPubKey},` AND I change in Validator Client implementation
the `params` variable, like in the following example:

```go
func (c beaconApiValidatorClient) getStateValidator(pubkey string) (*rpcmiddleware.StateValidatorsResponseJson, error) {
	// params := neturl.Values{"id": []string{pubkey}}
       params := neturl.Values{"<A WRONG KEY>": []string{pubkey}}

	url := buildURL(
		c.url,
		"eth/v1/beacon/states/head/validators",
		params,
	)
```

then the test will still success.

If I only modify the valdiator implementation without commenting the `[]string{stringPubKey},` line in the test, then the test will fail.

This `pubKeys` variable allow us to ensure the Validator Client calls the Beacon Node with the well suited URL.

Comment (nalepae) [reply validator/client/beacon-api/handlers_test.go:96]: Anyway, using GoMock instead.

File: validator/client/beacon-api/index.go
Comment (nalepae) [reply validator/client/beacon-api/index.go]: And vice versa!

Comment (nalepae) [reply validator/client/beacon-api/index.go]: @PatriceVignola I just tried to merge your `add-get-attestation-data` branch into this one for testing purpose.
Very minor (easily fixables) conflicts (mainly on `BUILD.bazel`, which is expected). 👌

Comment (nalepae) [reply validator/client/beacon-api/index.go]: I created a dedicated commit in this PR: https://github.com/prysmaticlabs/prysm/pull/11712/commits/52990d21348c055dfcf86be6cc077bc04113eb85
So it's easier for the review to separate your code from #11714 (which I copied here) from mine.

File: validator/client/beacon-api/index_test.go
Comment (rkapka) [inline validator/client/beacon-api/index_test.go:22]: this can be a `const`

File: validator/client/beacon-api/index.go
Comment (nalepae) [reply validator/client/beacon-api/index.go]: I finally implemented a generic `getStateValidators` function in [this commit](https://github.com/prysmaticlabs/prysm/pull/11712/commits/4adf2b343d108df8c9ee9c2d405733c35d8d4508).

**Review by rkapka — APPROVED** [review]

**Pull request merged**
